### ISSUE TITLE
sum quite slow compared to CPU

### ISSUE THREAD

**Issue by cdeterman** [viennacl/viennacl-dev#216]
I was doing some profiling on some code I had and the bottleneck came out to be when I was taking the absolute values and max of a vector.  I was using the following line:

```c++
viennacl::vector<T> vcl_A;
viennacl::linalg::max(viennacl::linalg::element_fabs(vcl_A));
```

Even with a length of 10,000 the performance is quite slow compared to CPU processing with something like the Eigen library.

```c++
Eigen::VectorXd A;
A.cwiseAbs().maxCoeff();
```

Has this been something previously noted as a current problem area or did I somehow miss something to make this more performant?

**Comment by karlrupp** [comment]
10k elements is still fairly small for GPUs - PCI-Express latency will dominate. Plus, you'll create a hidden temporary. Try `viennacl::linalg::norm_inf(vcl_A);` to avoid the temporary. For good performance you should further increase the vector size. 100k should be okay.

**Comment by cdeterman** [comment]
You are correct that `norm_inf` does help significantly but still slower than Eigen at 10K.  Also, once I get up to 100k it is at least beating out the base R speed but it still remains slower than Eigen.  Note, this is referring to when the data is already on the device (previously copied) compared to the CPU Eigen process.  There is no PCI-Express latency that should be involved here.
